powerauth-webflow-client/src/main/java/io/getlime/security/powerauth/app/webflow/demo/configuration/SecurityConfiguration.java

```diff
@@ -18,20 +18,106 @@
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.DefaultOAuth2AuthorizationRequestResolver;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestResolver;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
 
 /**
  * Default Spring Security configuration.
  *
  * @author Petr Dvorak, petr@wultra.com
+ * @author Roman Strobl, roman.strobl@wultra.com
  */
 @Configuration
-public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+@EnableWebSecurity
+public class SecurityConfiguration {
+
+    private final OAuth2AuthorizedClientRepository authorizedClientRepository;
+    private final OAuth2AuthorizedClientService authorizedClientService;
+    private final ClientRegistrationRepository clientRegistrationRepository;
+
+    @Autowired
+    public SecurityConfiguration(OAuth2AuthorizedClientRepository authorizedClientRepository, OAuth2AuthorizedClientService authorizedClientService, ClientRegistrationRepository clientRegistrationRepository) {
+        this.authorizedClientRepository = authorizedClientRepository;
+        this.authorizedClientService = authorizedClientService;
+        this.clientRegistrationRepository = clientRegistrationRepository;
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(final HttpSecurity http) throws Exception {
+        return http
+                .httpBasic().disable()
+                .authorizeHttpRequests(requests -> requests
+                        .requestMatchers("/", "/home", "/css/**", "/js/**", "/images/**").permitAll()
+                        .anyRequest().fullyAuthenticated()
+                )
+                .oauth2Login()
+                    .authorizedClientRepository(authorizedClientRepository)
+                    .authorizedClientService(authorizedClientService)
+                    .clientRegistrationRepository(clientRegistrationRepository)
+                    .authorizationEndpoint()
+                    .authorizationRequestResolver(new CustomAuthorizationRequestResolver(this.clientRegistrationRepository))
+                .and()
```

Comment:
It is possible to get rid of `and()` but we may postpone to a different pull request https://spring.io/blog/2019/11/21/spring-security-lambda-dsl